src/org/traccar/geocoder/NominatimGeocoder.java

```diff
@@ -16,14 +16,18 @@
 package org.traccar.geocoder;
 
 import javax.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class NominatimGeocoder extends JsonGeocoder {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(NominatimGeocoder.class);
+
     private static String formatUrl(String url, String key, String language) {
         if (url == null) {
-            url = "https://nominatim.openstreetmap.org/reverse";
+            url = "https://nominatim.openstreetmap.org/";
         }
-        url += "?format=json&lat=%f&lon=%f&zoom=18&addressdetails=1";
+        url += "reverse?format=json&lat=%f&lon=%f&zoom=16&addressdetails=1&extratags=1";
```

Comment:
Zoom 18 takes you to features such as gardens and parks, zoom 16 keeps you at street level.

Reply:
Does zoom 16 give you house numbers? That's the reason we are using 18.

Reply:
No, but in absence of house numbers and for devices with low accuracy, it will return inaccurate addresses, for example with my test device, was on a road but reported it was in a garden off of the road. Maybe the solution is to add a variable in the config to select this?

Reply:
Configuration sounds good to me.

Reply:
Will explore this at the weekend